### ISSUE TITLE
Improve handling of primitive typed comparison

### ISSUE THREAD

**Issue by andrewcraik** [eclipse/openj9#2833]
java/lang/Integer has a static compare function which takes two ints and does something like
if (x < y)
  return -1;
if (x == y)
  return 0;
return 1;
We should recognize this method and just compare x and y once and use the condition codes to compute the result (0, 1, -1). I believe there are similar methods for the other primitive integer types. This would probably be a recognized method, a dontInline if codegen supported, and a case in the codegens to handle the specific assembly sequence.

**Comment by andrewcraik** [comment]
FYI @0dvictor since you work in this area a lot

**Comment by 0dvictor** [comment]
Good idea. I actually thought about this a while ago and had some experiments. Let me make polish my prototype and create a pull request.

**Comment by 0dvictor** [comment]
I got a prototype working and tested. However, during this process I got a question that I would like to clarify before creating a pull request.

Quote from Java 8 Specification:
> public static int compare(int x, int y)
> ...
> Returns: the value 0 if x == y; a value less than 0 if x < y; and a value greater than 0 if x > y

The specification does not require returning -1, 0, or 1; although the JCL implementation may do.

So I got two questions:
1. Should the CodeGen generate equivalent code to the JCL implementation, or it can generate any result as long as complying the Specification?
2. If the CodeGen need comply with the JCL implementation, may the Optimizer (VP) relying the behavior of the JCL implementation performing certain optimizations? i.e. create VP constraint [-1, 1].
